ops/framework.py

```diff
@@ -88,7 +88,48 @@ def from_path(cls, path):
         return handle
 
 
-class EventBase:
+class UnassignedEvent(type):
+    """Metaclass for events which have not yet been assigned to an emitter type.
+
+    Without being assigned and bound to an emitter, unassigned events cannot be emitted.
+    """
```

Comment:
This is actually the metaclass of the event. Assigned, unassigned, or whatever else happens, it's still its metaclass. It's also a single line long, and never going to be used directly by developers, so no need to document it in that style. Accordingly, its name should indicate what it really is. I suggest `EventMetaType`.

Reply:
Following what's going on when 2 Python features (`__set_name__` and descriptors) are involved is a bit difficult without either helpful naming or a docstring in my view.

By glancing over at names at first I could not tell the difference between `Assigned` and `Bound`.

If I read out loud what's going on:

- Assigning a subclass of `UnassignedEvent` to a class attribute will instead result in an assignment of an event descriptor to that attribute;
- When that attribute is then accessed and the descriptor's `__get__` method is invoked, an instance of `BoundEvent` is returned which can then be used to emit events tied to an emitter type.

I think it should be clear that we aim to bind "unbound" event types to another type. This is done via descriptors so `EventDescriptor` sounds like a good name.

`UnboundEvent` instead of `UnassignedEvent` correlates with `BoundEvent` and makes the intention a bit clearer to me.

`EventMetaType` would also work so long as there is an explanation about the required binding in the docstring.

Reply:
I went with `EventMetaType` as suggested and a hopefully short but useful doc string.